builtins/match.go

```diff
@@ -34,12 +38,76 @@ func matchRecord(a, b *runtime.Record) (bool, error) {
 			}
 		} else {
 			return false, runtime.Errorf("Value mismatch: field %s in second Record not found in first", k)
-
 		}
 	}
 	return true, nil
 }
 
+func matchRecordOf(a, b *runtime.List) (bool, error) {
```

Comment:
You can improve the readability of this function by reducing the number of variable declarations. Also naming the value slice `str` is a bit misleading. I'd find something like `s` or `val` easier. I suggest you change the signature to:
```suggestion
func matchRecordOf(val, pat []runtime.Object) (bool, error) {
```

and call it with `matchRecordOf(a.(*runtime.List).Elements, b.Elements)` directly.